src/main/proto/spine/options.proto

```diff
@@ -127,24 +127,92 @@ extend google.protobuf.FieldOptions {
+    //      // having `(validate)` constraint that forces the message to meet its constraints.
+    //      // Since the `address` field is optional for `Student`, `(validate)` would allow
+    //      // default instances for this field treating them as "no value set".
+    //      message Student {
+    //          Address address = 1 [(validate) = true]; // implicit `(required) = false`.
+    //      }
+    //
+    //      // Make the validated field required to avoid this behavior. In this case, `(validate)`
+    //      // continues to bypass default instances, but the `(required)` option will report them.
+    //      message Student {
+    //          Address address = 1 [(validate) = true, (required) = true];
+    //      }
+    //      ```
+    //
+    //   2. For repeated fields: every element in the repeated field must meet the constraints
+    //      of its message type.
+    //
+    //      Example:
+    //
+    //      ```
+    //      // Note that the default instance of `PhoneNumber` is not valid because the `value`
+    //      // field is mandatory.
+    //      message PhoneNumber {
+    //          string value = 1 [(required) = true, (pattern).regex = "^\+?[0-9\s\-()]{1,30}$"];
+    //      }
+    //
+    //      // In contrast to singular fields, the default instances in `repeated` will also be
+    //      // reported by the `(validate)` constraint, with those that do not match the pattern.
+    //      message Student {
+    //          repeated PhoneNumber number = 1 [(validate) = true];
+    //      }
+    //      ```
+    //
+    //   3. For map fields: each value in the map must meet the constraints of message type.
+    //      Note: Protobuf does not allow messages to be used as map keys.
+    //
+    //      Example:
+    //
+    //      ```
+    //      // Note that the default instance of `PhoneNumber` is not valid because the `value`
+    //      // field is mandatory.
+    //      message PhoneNumber {
+    //          string value = 1 [(required) = true, (pattern).regex = "^\+?[0-9\s\-()]{1,30}$"];
+    //      }
+    //
+    //      // In contrast to singular fields, the default instances in `map` values will also be
+    //      // reported by the `(validate)` constraint, with those that do not match the pattern.
+    //      message Contacts {
+    //          map<string, PhoneNumber> map = 1 [(validate) = true];
+    //      }
+    //      ```
+    //
+    // If the field contains `google.protobuf.Any`, the option will first attempt to unpack
+    // the enclosed message, and only then validate it. However, unpacking is not always possible:
+    //
+    //   1. The default instance of `Any` is always valid because there is nothing to unpack
+    //      and validate.
+    //   2. Instances with unknown type URLs are considered valid because their Java class cannot
+    //      be determined from `io.spine.KnownTypes`.
```

Comment:
On this level of documentation, mentioning `KnownTypes` triggers more questions than explains the state of matters.

Please try to explain it, as if the readers aren't aware of the internal machinery under the Spine's hood.